pkg/engine/internal/planner/physical/planner.go

```diff
@@ -155,6 +152,10 @@ func (p *Planner) process(inst logical.Value, ctx *Context) ([]Node, error) {
 		return p.processVectorAggregation(inst, ctx)
 	case *logical.Parse:
 		return p.processParse(inst, ctx)
+	case *logical.BinOp:
+		return p.processBinOp(inst, ctx)
+	case *logical.UnaryOp:
```

Comment:
I don't think this branch can ever be reached.

The logical plan does not create UnaryOp outside of a Predicate, and the physical planner uses the `convertPredicate()` logic for that. The other case is a UnaryOp as math expression, which is only ever a child of a BinOp. And then it is processed in `processMathExpressionChild()`.